packages/vit-s/src/hooks.js

```diff
@@ -322,3 +322,58 @@ export function useGetObsInfo(obsType, obsLabelsTypes, obsLabelsData, obsSetsMem
     return null;
   }, [obsType, obsLabelsTypes, obsLabelsData, obsSetsMembership]);
 }
+
+
+export const useMappedGeneList = (geneObject, featureMap) => {
+  const [mappedGeneList, setMappedGeneList] = useState(geneObject);
+  useEffect(() => {
+    const processFeatureLabelMap = async () => {
+      // Assuming featureMap is set for symbolic genes
+      if (featureMap !== undefined) {
+        setMappedGeneList(geneObject);
+        return;
+      }
+      try {
+        let ensbGenes = false;
+        if (Array.isArray(geneObject)) ensbGenes = geneObject?.some(value => value?.toUpperCase().startsWith('ENSG'));
+        else {
+          ensbGenes = Object.entries(geneObject)?.some(([key, value]) => key.toLowerCase().includes('gene') && value?.toUpperCase().startsWith('ENSG'));
+        }
```

Comment:
Personally, I'm not a huge fan of single-line `if` statements without curly braces in terms of readability, but that's a matter of preference 😄

Personal preference aside, the if/else statement itself is fine, but I think we could improve readability/understandability.

I think extracting the filtering functions out into variables would be helpful; this would also allow us to switch the if/else into a ternary to make it more compact while keeping things readable.
```
const isEnsemblGene = value => value?.toUpperCase().startsWith('ENSG');
const isGeneKey = key => key.toLowerCase().includes('gene')

const ensbGenes = Array.isArray(geneObject) 
	? geneObject.some(isEnsemblGene) 
	: Object.entries(geneObject).some(([key, value]) => isGeneKey(key) && isEnsemblGene(value))
```